erpnext/accounts/doctype/payment_manager/payment_manager.py

```diff
@@ -0,0 +1,429 @@
+			if (mapped_item is not None):
+				bank_entry.party_type = mapped_item.mapped_data_type
+				bank_entry.party = mapped_item.mapped_data
+			else:
+				bank_entry.party_type = "Supplier" if not entry["Deposits"].strip() else "Customer"
+				party_list = frappe.get_all(bank_entry.party_type, fields=["name"])
+				parties = [party.name for party in party_list]
+				matches = difflib.get_close_matches(bank_entry.description.lower(), parties, 1, 0.4)
+				if len(matches) > 0: bank_entry.party = matches[0]
+			bank_entry.amount = -float(entry["Withdrawals"]) if not entry["Deposits"].strip() else float(entry["Deposits"])
+		self.map_unknown_transactions()
+		self.map_transactions_on_journal_entry()
+
+	def map_transactions_on_journal_entry(self):
+		for entry in self.new_transaction_items:
+			vouchers = frappe.db.sql("""select name, posting_date from `tabJournal Entry`
+										where posting_date='{0}' and total_credit={1} and cheque_no='{2}' and docstatus != 2
+									""".format(datetime.strptime(entry.transaction_date, "%Y-%m-%d").date(), abs(entry.amount), entry.description), as_dict=True)
+			if (len(vouchers) == 1):
+				entry.reference_name = vouchers[0].name
+
+	def populate_matching_invoices(self):
+		self.payment_invoice_items = []
+		self.map_unknown_transactions()
+		added_invoices = []
+		for entry in self.new_transaction_items:
+			if (not entry.party or entry.party_type == "Account"): continue
+			account = self.receivable_account if entry.party_type == "Customer" else self.payable_account
+			invoices = get_outstanding_invoices(entry.party_type, entry.party, account)
+			transaction_date = datetime.strptime(entry.transaction_date, "%Y-%m-%d").date()
+			outstanding_invoices = [invoice for invoice in invoices if invoice.posting_date <= transaction_date]
+			amount = abs(entry.amount)
+			matching_invoices = [invoice for invoice in outstanding_invoices if invoice.outstanding_amount == amount]
+			sorted(outstanding_invoices, key=lambda k: k['posting_date'])
+			for e in (matching_invoices + outstanding_invoices):
+				added = next((inv for inv in added_invoices if inv == e.get('voucher_no')), None)
+				if (added is not None): continue
+				ent = self.append('payment_invoice_items', {})
+				ent.transaction_date = entry.transaction_date
+				ent.payment_description = entry.description
+				ent.party_type = entry.party_type
+				ent.party = entry.party
+				ent.invoice = e.get('voucher_no')
+				added_invoices += [ent.invoice]
+				ent.invoice_type = "Sales Invoice" if entry.party_type == "Customer" else "Purchase Invoice"
+				ent.invoice_date = e.get('posting_date')
+				ent.outstanding_amount = e.get('outstanding_amount')
+				ent.allocated_amount = min(float(e.get('outstanding_amount')), amount)
+				amount -= float(e.get('outstanding_amount'))
+				if (amount <= 5): break
```

Comment:
should this be a setting?

Reply:
A configuration is perhaps overkill, I will look at completely removing it. I guess I was using that to avoid minor round-off issues, which certainly needs to be configurable if we are serious about it.